### ISSUE TITLE
-Wshadow false negative for function parameter shadowing (partially; incorrectly) a global variable (incorrectly used as array length expression)

### ISSUE THREAD

**Issue by alejandro-colomar** [llvm/llvm-project#128831]
```c
alx@debian:~/tmp$ cat f.c
extern int n;

void f(int a[n], int n);
alx@debian:~/tmp$ gcc -Wall -Wextra -Wshadow -S f.c 
alx@debian:~/tmp$ clang -Weverything -Wno-vla -S f.c 
alx@debian:~/tmp$ 
```

That program obviously has a bug.  The number of elements of 'a' is intended to be $2, but it is actually controlled by the global variable 'n'.  I think -Wshadow should diagnose this.
